### DISCUSSION TITLE
Flexible query for client side JS

### DISCUSSION THREAD

**Discussion by IRediTOTO** [supabase/supabase#1309] [Questions] [answered]
Hi, I just jump into Supabase and fall in love with it.
Im reading docs but dont you have a flexible syntax like this? 

```
const getAllUsers=await supabaseAdmin.sql(`
select * from "auth.users"
`)
// I just want query all users data but don't know how to do :c

```
I think use directly SQL syntax is more easier than read the docs and try to find what syntax do.

**Comment by marcusguttenplan** [comment]
You should be able to hit postgres directly with a SQL package! We pointed part of our Rails app to Supabase and it can read/write with its own ORM and Supabase still picks up updates. It's really sweet!

In the settings it gives you connection info for the db that should work

**Reply by IRediTOTO** [reply]
I got it, thank you. But can you tell me how to query all users email with supabase-js?

**Comment by silentworks** [comment, marked as answer]
Your suggestion would be harder to work with for the Supabase team and also most developers who are not so familiar with SQL would struggle to use it. It would be harder for the Supabase team because they would then have to convert that SQL string into a RESTful endpoint to make a request to Supabase Database. If they were to pass the SQL string down directly they would have to do more checks to make sure the code isn't malicious. You can create a view to return only the data you require from the `auth` schema inside of the `public` schema, but do be careful with this or you could expose sensitive information.

Inside of the SQL editor inside of Supabase create the view with the code below:

```sql
CREATE VIEW "public"."user_emails" AS SELECT auth.users.id,
    auth.users.email
   FROM auth.users;
```

Inside of your supabase-js do the following:

```js
const getAllUsersEmails = await supabase.from(`users_emails`).select('*');
```

Again I'm going to emphasise that you don't want to expose sensitive information from the `auth.users` table to your application. You could have also accomplished this by having a stored procedure that would copy the user's email into a `public.users` table and query that instead.

**Reply by IRediTOTO** [reply]
Hi, thank you for your solution. Yes I understand your concern. But sometime we need to find some data like users not active for a long time, users with name "abc", get all users from list id from others tables. I think this is just a basic query. I am not good at SQL, but i can search Google for it. I just can't find how to make it work with client js. I query secret data on server use bypass api, not in browser.

**Reply by silentworks** [reply]
The approach I would take to do the sort of queries you are after is the above, I would just create a bunch of views matching those queries, do remember you can still use the normal filtering and selection on a view as you would on a normal table.

```js
await supabase.from('users_emails').select('email'); // this should only return the email column of that view
```
